src/main/java/net/server/channel/handlers/PlayerLoggedinHandler.java

```diff
@@ -409,7 +408,14 @@ public final void handlePacket(SeekableLittleEndianAccessor slea, MapleClient c)
                 }
                 
                 if (YamlConfig.config.server.USE_NPCS_SCRIPTABLE) {
-                    c.announce(MaplePacketCreator.setNPCScriptable(ScriptableNPCConstants.SCRIPTABLE_NPCS));
+                    List<Integer> npcsIds = YamlConfig.config.server.NPCS_SCRIPTABLE;
+
+                    // Any npc be specified as the rebirth npc. Allow the npc to use custom scripts explicitly.
+                    if (YamlConfig.config.server.USE_REBIRTH_SYSTEM) {
+                        npcsIds.add(YamlConfig.config.server.REBIRTH_NPC_ID);
+                    }
```

Comment:
You are mutating the global list of npcsIds. Each time a player logs in and the rebirth system is enabled, the rebirth npc id will be appended, so after a while, when a bunch of players have logged in, there will be a bunch of duplicate npc ids at the end of the list.
You should do something like:
`List<Integer> npcsIds = new ArrayList(YamlConfig.config.server.NPCS_SCRIPTABLE);` on line 411

Reply:
Thanks, should be fixed.